dask_sql/physical/rel/logical/window.py

```diff
@@ -334,9 +342,8 @@ def _apply_window(
 
         for c in newly_created_columns:
             # the fields are in the correct order by definition
-            field_name = field_names[len(cc.columns)]
+            field_name = field_names[len(cc.columns) - input_column_count]
             cc = cc.add(field_name, c)
```

Comment:
To be clear on my last comment, I'm thinking about something like

```suggestion
            field_index = len(cc.columns) - input_column_count
            field_name = field_names[field_index]
            # would insert into `cc,_frontend_columns` at `field_index`
            cc = cc.add(field_name, field_index, c)
```

Reply:
I'm in favor of this. Being able to insert at a different spot than the end would simplify having to use `limit_to` which isn't limiting the number of columns in this case but just shuffling the order around. Happy to tackle in a followup pr